Kogito 0.8.0 release (#6)
kiegroup/kogito-travel-agency-tutorial: 08-kogito-travel-agency/README.md

* Destination that the traveller wants to go - country
* Nationality of the traveller
* Length of the stay

The result will be “yes” or “no”.

<p align="center"><img width=75% height=50% src="docs/images/decisiontable.png"></p>

## Business logic

Business logic will be based on business processes

Public process that will be responsible for orchestrating complete travel request

<p align="center"><img width=75% height=50% src="docs/images/travels-process.png"></p>

Private process that will be responsible for booking a hotel.

<p align="center"><img width=75% height=50% src="docs/images/book-hotel-process.png"></p>

Private process that will be responsible for booking a flight.

<p align="center"><img width=75% height=50% src="docs/images/book-flight-process.png"></p>

## Services

There will be services implemented to carry on the hotel and flight booking. Implementation will be a CDI beans that will have hard coded logic to return a booked flight or hotel.

* org.acme.travels.service.HotelBookingService
* org.acme.travels.service.FlightBookingService

# Try out the complete service

## Installing and Running

### Prerequisites

You will need:

>   - Java 1.8.0+ installed

Reviewer: ```suggestion
  - Java 11+ installed
```

  - Environment variable JAVA_HOME set accordingly
  - Maven 3.6.2+ installed

When using native image compilation, you will also need:
  - GraalVM installed
  - Environment variable GRAALVM_HOME set accordingly
  - Note that GraalVM native image compilation typically requires other packages (glibc-devel, zlib-devel and gcc) to be installed too, please refer to GraalVM installation documentation for more details.

### Infrastructure requirements

#### Infinispan

This application requires an Infinispan server to be available and by default expects it to be on default port and localhost.

> You can install Infinispan server by downloading it from [official website](https://infinispan.org/download) version to be used in 10.0.0.CR1

Reviewer: ```suggestion
You can install Infinispan server by downloading it from [official website](https://infinispan.org/download) version to be used in 10.1.3.Final
```

tables and java code. No need to redeploy or restart your running application.During this workshop we will create a software system for a startup travel agency called Kogito Travel Agency. The first iteration of the system will consist of a set of services that are able to deal with travel requests and the booking of hotels and flights.

### Compile and Run using Local Native Image
Note that this requires GRAALVM_HOME to point to a valid GraalVM installation

```
mvn clean package -Pnative
```

To run the generated native executable, generated in `target/`, execute

```
./target/kogito-travel-agency-{version}-runner
```

### Start Kogito Data Index Service

We provide a `startDataIndex.sh` and `startDataIndex.ps1` script in the `scripts` directory of this tutorial. This script will

* Check that the **Kogito Travel Agency** and **Kogito Visas** projects have been compiled and that the required protobuf files are available.
* Copy the protobuf files to a local directory.
* Download the **Kogito Data Index Service Runner** if required.
* Start the **Kogito Data Index Service Runner** with the copied protobuf files.

> If you wish to install, configure and start the **Data Index Service** manually, the _runnner_ can be downloaded from [Kogito Data Index Service 0.8.0](https://search.maven.org/remotecontent?filepath=org/kie/kogito/data-index-service/0.8.0/data-index-service-0.8.0-runner.jar)

Reviewer: 0.8.1

NOTE: Other versions can be found here [Kogito Data Index Service versions](https://search.maven.org/artifact/org.kie.kogito/data-index-service)

This service works with .proto (protobuf) files that define the data model. fter downloading the runner, create a new folder to store the .proto files that will be used by the service,  e.g. `persistence`. Copy the protobuf files from the **Kogito Travel Agency** and **Kogito Visas** application to the `persistence` folder. These files can be found in the `target/classes/persistence` folders of these projects.

To start the **Kogito Data Index Service**, execute the Java runner JAR of the service, providing it the location of the protobuf files:

```

> java -jar  -Dkogito.protobuf.folder={full path to proto files folder} data-index-service-0.4.0-runner.jar

Reviewer: ```suggestion
java -jar  -Dkogito.protobuf.folder={full path to proto files folder} data-index-service-0.8.1-runner.jar
```

```

e.g.

```

> java -jar  -Dkogito.protobuf.folder=`pwd`/persistence data-index-service-0.4.0-runner.jar

Reviewer: ```suggestion
java -jar  -Dkogito.protobuf.folder=`pwd`/persistence data-index-service-0.8.1-runner.jar
```

```

### Start Kogito Management Console

The Kogito Management Console consists of a frontend and backend component. These 2 components can be combined into a single executable runner using the project's Maven build.

To package the frontend and backend together, use the following Maven build command: `mvn clean package -Dui.deps -Dui`
This will:
* Install the required front-end dependencies.
* Build the front-end,
* Compile the back-end.
* Package the front-end and back-end into a single Java runner.

> After the build has completed, you can start the Manaagement Console with the following command: `java -jar target/management-console-0.8.0-runner.jar`

Reviewer: 0.8.1
